penguin/integrations/mcp/server.py

```
@@ -1,124 +1,361 @@
-from typing import Any, Dict, Iterable, List, Optional, Tuple
+import logging
+import re
+import sys
+from dataclasses import dataclass, field
+from typing import Any, Callable, Iterable, Optional
+
+try:  # pragma: no cover - import availability depends on optional extra
+    from mcp.server.fastmcp import FastMCP
+
+    HAS_MCP_SERVER_SDK = True
+except Exception:  # pragma: no cover - exercised when optional extra missing
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)